### ISSUE TITLE
Allow importing CSS / SCSS into JS on server

### ISSUE THREAD

**Issue by clayne11** [meteor/meteor#6852]
I'm not sure if this is supposed to be possible, but it would certainly be extremely nice if it was.

As of 1.3 we now have proper ES6 import / export and Npm support. One thing I'm still unable to do, however, is to import CSS / SCSS into JS. Example:

```
import React, {PropTypes} from 'react';

import './tester.scss';

const Tester = ({}) => (
  <div className='Tester'>
  </div>
);

export default Tester;
```

When I import `./tester.scss`, Meteor fails to build with `Exited with code: 1`. Same thing happens if its a `css` file. Is this feature not supported at this time?

It would be really, really nice for refactoring packages and general development speed and clarity to be able to import a style file into it's JS counterpart.

Am I missing something or is this simply not possible at the moment?

**Comment by sebakerckhof** [comment]
This should be possible with meteor 1.3.2, but the scss/css file needs to be in an `/imports` folder.

**Comment by clayne11** [comment]
All my SCSS files are in packages. I built my app into a package-based architecture before Meteor 1.3 came out. Is it not possible to do in packages?

**Comment by sebakerckhof** [comment]
Seems to work for me, here's an example repo: https://github.com/sebakerckhof/scssimports

**Comment by clayne11** [comment]
You only imported your component on the client.

To allow for SSR you should be importing the JS file both on the client and the server, and that's where the issue occurs.

**Comment by clayne11** [comment]
You only import your component on the client in the example repo.

For SSR the component needs to be imported on both the client and the server, and that's where the issue occurs.

**Comment by clayne11** [comment]
I just looked at the code for [`addStyles`](https://github.com/meteor/meteor/blob/devel/packages/modules/css.js) and it's clear to me now why this isn't working.

How are we intended to write universal apps when this pattern isn't permitted?

**Comment by clayne11** [comment]
This is actually a serious issue. I migrated to importing SCSS only in the client for all of my React components and now I've noticed that the CSS is no longer going through the minification and PostCSS pipeline and I'm not getting autoprefixer added to the resultant CSS.

**Comment by abernix** [comment]
@clayne11 Pretty sure this is the expected behaviour, as [explained in the Meteor Guide](http://guide.meteor.com/using-packages.html#using-npm).

> When importing CSS from a JavaScript file that CSS is not bundled with the rest of the CSS processed with the Meteor Build tool, but instead is put in your app’s <head> tag inside <style>...</style> after the main concatenated CSS file.

Basically, when importing it directly, you're bypassing that functionality that the `meteor` tool provides.

**Comment by sebakerckhof** [comment]
The problem is that meteor doesnt allow multiple compilers for the same
file type or to chain them. Therefore we've put autoprefixing in the minify
step, which isn't run when importing via js.
On May 2, 2016 11:27, "Jesse Rosenberger" <notifications@github.com> wrote:

@clayne11 <https://github.com/clayne11> Pretty sure this is the expected
behaviour, as explained in the Meteor Guide
<http://guide.meteor.com/using-packages.html#using-npm>.

When importing CSS from a JavaScript file that CSS is not bundled with the
rest of the CSS processed with the Meteor Build tool, but instead is put in
your app’s tag inside ... after the main concatenated CSS file.

Basically, when importing it directly, you're bypassing that functionality
that the meteor tool provides.

—
You are receiving this because you commented.
Reply to this email directly or view it on GitHub
<https://github.com/meteor/meteor/issues/6852#issuecomment-216170672>

**Comment by clayne11** [comment]
I switched over to `nathantreid:css-modules` which has the `PostCSS` integration built into it's own pipeline which has solved this issue for me.

I still find it rather annoying that I can't use `import` with styles on the server though, because when I use `require` then `eslint-plugin-import` doesn't lint my import which is surface area for a bug.

I would love if `addStyles` on the server was simply a no-op so I could go back to `import`ing styles both on the client and the server.

**Comment by clayne11** [comment]
Any reason why we can't simply make `import 'foo.scss'` a no-op on the server so that we can use `import` for styles both on the client and server?

**Comment by lope** [comment]
I noticed this worked in 1.4 beta 9 (only the styles I imported got loaded) but then not in 1.4 beta 14. I've not tried with the new 1.4 release.

**Comment by hwillson** [comment]
To help provide a more clear separation between feature requests and bugs, and to help clean up the feature request backlog, Meteor feature requests are now being managed under the https://github.com/meteor/meteor-feature-requests repository. 

This feature request will be closed here, but anyone interested in migrating this feature request to the new repository (to make sure it stays active), can click [here](https://github.com/meteor/meteor-feature-requests/issues/new?title=Allow%20importing%20CSS%20%2F%20SCSS%20into%20JS%20on%20server&body=Migrated%20from:%20meteor/meteor%236852) to start the feature request migration process. This manual migration process is intended to help identify which of the older feature requests are still considered to be of value to the community. Thanks!
